test/PerunPlutus/TestCases.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE NumericUnderscores #-}
+module PerunPlutus.TestCases where
+import Plutus.Contract.Test
+import PerunPlutus.PerunDummySpec
+import Plutus.Contract.Test.ContractModel
+import PerunDummy
+import Test.QuickCheck
+import Data.Tuple.Extra (both)
+
+-- Testcases
+
+-- unitTest example of creating a channel, updating it once and closing.
+honestPaymentTest :: (Wallet, Wallet, Wallet) -> DL PerunModel ()
+honestPaymentTest (wa, wb, wf) = do
+  (initBalA, initBalB) <- forAllQ $ both chooseQ ((5_000_000, 20_000_000), (5_000_000, 20_000_000))
+  action $ Open wf (wa, wb) 42069 initBalA initBalB 15000
+
+  modChSt <-
+    getContractState >>= \case
+      PerunModel Nothing -> fail "opening a channel should be tracked in PerunModel"
+      PerunModel (Just (cs, _)) -> aPaysB cs 420
+  action $ Update modChSt
+
+  action Finalize
+
+  getContractState >>= \case
+    PerunModel Nothing -> fail "unclosed channel should persist in PerunModel"
+    PerunModel (Just (ChannelState cid ba bb v _, _)) -> action $ Close wb (wa, wb) cid ba bb v
+
+singleDisputeTest :: (Wallet, Wallet, Wallet) -> DL PerunModel ()
+singleDisputeTest (wa, wb, wf) = do
+  (initBalA, initBalB) <- forAllQ $ both chooseQ ((5_000_000, 20_000_000), (5_000_000, 20_000_000))
+  action $ Open wf (wa, wb) 42069 initBalA initBalB 15000
+
+  modChSt <-
+    getContractState >>= \case
+      PerunModel Nothing -> fail "opening a channel should be tracked in PerunModel"
+      PerunModel (Just (cs, _)) -> aPaysB cs 420
+  action $ Update modChSt
+
+  getContractState >>= \case
+    PerunModel Nothing -> fail "unclosed channel should persist in PerunModel"
+    PerunModel (Just _) -> action $ Dispute wb (wa, wb) 42069 modChSt
+  action $ Wait 16
```

Comment:
I propose the following:

We should always, unless impossible, have `action`s on the same indentation level.
So instead of immediately executing the action here using `ChannelState {..}`, we would `return cs` and pattern match on the same level where all `action`s are.

```haskell
  modChSt <-
    requireWithChannel
      "channel must be available after opening"
      (\(cs, _) -> forAllQ (chooseQ (0, initBalLB)) >>= aPaysB cs)
  action $ Update modChSt
  action $ Dispute wb (wa, wb) channelID modChSt
  action $ Wait 16
  action $ ForceClose wb (wa, wb) channelID
```

This makes the tests easier to read, IMHO. You can look at the way I restructured the existing tests in [my PR](https://github.com/perun-network/perun-cardano-contract/pull/7/files#diff-00dc6a7d2614e6bd605b6965ac2bd0437d9f9dbfad218559db87907e14f07938)

Of ourse I do not claim to have be the overmind with the most rational argumentation, so feel free to express your opinions (:

Just browse through the `TestCases` file. If you agree with the changes you can merge them into your own branch. I also added rudimentary test discovery, it still requires some manual labor, but AFAICT this is the qay `QuickCheck` is designed...